var/spack/repos/builtin/packages/cppad/package.py

```diff
@@ -9,20 +9,38 @@
 class Cppad(CMakePackage):
     """A Package for Differentiation of C++ Algorithms."""
 
-    homepage = "https://www.coin-or.org/CppAD/"
-    url = "http://www.coin-or.org/download/source/CppAD/cppad-20170114.gpl.tgz"
+    maintainers("whart222")
+
+    homepage = "https://github.com/coin-or/CppAD"
+    url = "https://github.com/coin-or/CppAD/archive/refs/tags/20240000.4.tar.gz"
     git = "https://github.com/coin-or/CppAD.git"
 
-    version("develop", branch="master")
+    version("master", branch="master")
+    version(
+        "20240000.4", sha256="0dfc1e30b32d5dd3086ee3adb6d2746a019e9d670b644c4d5ec1df3c35dd1fe5"
+    )
     version(
-        "20180000.0", sha256="1c355713e720fc5226ff3d6db2909922d46cd7ee0d36ee7985882f86905f655a"
+        "20220000.5", sha256="9fb4562f6169855eadcd86ac4671593d1c0edf97bb6ce7cbb28e19af2bfc165e"
+    )
+    version(
+        "20180000.0",
+        sha256="1c355713e720fc5226ff3d6db2909922d46cd7ee0d36ee7985882f86905f655a",
```

Comment:
I still debate whether this should be changed to the new sha256 despite the deprecation:

```suggestion
        sha256="a5226e4c5aa2ad6d95668f987b39939315bf134a0a793231984e6d42d6488cca",
```

Reply:
I'm good with this change.  I just pushed this fix.